creator/files/mutations/version.py

```diff
@@ -219,8 +224,82 @@ def mutate(
         return VersionUploadMutation(success=True, version=version)
 
 
+class EvaluateTemplateMatchInput(graphene.InputObjectType):
+    """Parameters used when validating a file version against templates"""
+
+    file_version = graphene.ID(
+        required=True, description="The file version being validated"
+    )
+    study = graphene.ID(
+        required=True,
+        description="The study which the file version belongs to"
+    )
+
+
+class EvaluateTemplateMatchMutation(graphene.Mutation):
+    """
+    Evaluate a file version against the templates in the study that the
+    file belongs to. For each template, report the details of the mismatch
+    between the file's columns and the template's required and optional
+    columns.
+
+    See creator.files.utils.evaluate_template_match for details on the match
+    evaluation
+    """
+    class Arguments:
+        input = EvaluateTemplateMatchInput(
+            required=True,
+            description="Attributes for the mutation",
+        )
+
+    results = graphene.List(TemplateMatchResult)
+
+    def mutate(self, info, input):
```

Comment:
Just realized I should check if user has permission to view file and study in this mutation. Will implement once I return to my comp